### DISCUSSION TITLE
Is it possible to skip execution of empty command

### DISCUSSION THREAD

**Discussion by Jai-JAP** [akinomyoga/ble.sh#420] [Q&A] [answered]
Currently (Atleast by default) its not possible to skip execution of empty commands (or just spaces) & it simply gives an empty prompt on pressing <kbd>RET</kbd>.

Is there a way to skip execution of the current command if it is empty (or just whitespaces) when <kbd>RET</kbd> is pressed?

**Comment by akinomyoga** [comment, marked as answer]
It's already skipping the execution of the command. It just reprints the prompt in the current implementation. If you want to skip the reprinting, you can modify the behavior of `ble/widget/default/accept-line`. For example,

```bash
# blerc

ble/function#advice around ble/widget/default/accept-line '
  ble-edit/content/clear-arg
  [[ ${_ble_edit_str//["$_ble_term_IFS"]} ]] && ble/function#advice/do'
```

**Reply by Jai-JAP** [reply]
That hopefully solves my query. I'll try that today.

**Reply by Jai-JAP** [reply]
Ok I have used this for some days now and in the current setup the prompt reappears on pressing <kbd>CTRL</kbd>+<kbd>C</kbd>

Is it possible to fix that?

**Reply by akinomyoga** [reply]
You can just do the same for `discard-line`:

```bash
ble/function#advice around ble/widget/discard-line '
  ble-edit/content/clear-arg
  [[ ${_ble_edit_str//["$_ble_term_IFS"]} ]] && ble/function#advice/do'
```

Or if you don't want to make `C-c` do anything, you can disable it by

```bash
ble-bind -f C-c nop
```

**Reply by Jai-JAP** [reply]
Ok that works.
i just looked around for a similar thing in `ble/widget/default` and couldn't find one.

Are both accept-line widgets same in `ble/widget/default` and `ble/widget`?

**Reply by akinomyoga** [reply]
> Are both accept-line widgets same in `ble/widget/default` and `ble/widget`?

No. `ble/widget/accept-line` performs extra work depending on the current keymap. `ble/widget/default/accept-line` is called by `ble/widget/accept-line`. That extra work is needed even for the empty command line. What we would like to skip this time is just `ble/widget/default/accept-line`, so we need to hook into `ble/widget/default/accept-line` but not into `ble/widget/accept-line`.
